presto-parser/src/main/java/io/prestosql/sql/parser/AntlrATNCacheFields.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.sql.parser;
+
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.Parser;
+import org.antlr.v4.runtime.atn.ATN;
+import org.antlr.v4.runtime.atn.LexerATNSimulator;
+import org.antlr.v4.runtime.atn.ParserATNSimulator;
+import org.antlr.v4.runtime.atn.PredictionContextCache;
+import org.antlr.v4.runtime.dfa.DFA;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class AntlrATNCacheFields
+{
+    private final ATN atn;
+    private final PredictionContextCache predictionContextCache;
+    private final DFA[] decisionToDFA;
+
+    public AntlrATNCacheFields(ATN atn)
+    {
+        this.atn = requireNonNull(atn, "atn is null");
+        this.predictionContextCache = new PredictionContextCache();
+        this.decisionToDFA = createDecisionToDFA(atn);
+    }
+
+    public void configureLexer(Lexer lexer)
+    {
+        requireNonNull(lexer, "lexer is null");
+        // Intentional identity equals comparison
+        if (this.atn != lexer.getATN()) {
+            throw new IllegalArgumentException(format("Lexer ATN mismatch: expected %s, found %s", this.atn, lexer.getATN()));
```

Comment:
Can we use `checkArgument` for this? The parser depends on Guava already.

Reply:
Re: checkArgument- I’m avoiding it here because it would incur boxing on the happy / fast path since we have more than 1 argument to interpolate.

Re: the use of “this”, I agree in this form it’s become less necessary (refactored code) but doesn’t seem to hurt given the cryptic nature of these field names and the reliance on object identity for correctness.

Reply:
There's actually an overload for the two argument `Object` version Guava 20

Reply:
Ah, didn't realize will change.